libevm/legacy/legacy.go

```diff
@@ -31,7 +35,10 @@ func (c PrecompiledStatefulContract) Upgrade() vm.PrecompiledStatefulContract {
 	return func(env vm.PrecompileEnvironment, input []byte) ([]byte, error) {
 		gas := env.Gas()
 		ret, remainingGas, err := c(env, input, gas)
-		if used := gas - remainingGas; used < gas {
+		if remainingGas > gas {
+			return nil, fmt.Errorf("remaining gas %d exceeds supplied gas %d", remainingGas, gas)
+		}
+		if used := gas - remainingGas; used <= gas {
```

Comment:
```suggestion
		if used := gas - remainingGas; used > 0 {
```

This was originally a (bad) check for the above, using overflow instead of an explicit comparison. Now that the comparison is in place, we can have a clearer condition.